src/diffusers/models/controlnet_hunyuan.py

```diff
@@ -0,0 +1,352 @@
+            timestep, encoder_hidden_states_t5, image_meta_size, style, hidden_dtype=timestep.dtype
+        )  # [B, D]
+
+
+        # text projection
+        batch_size, sequence_length, _ = encoder_hidden_states_t5.shape
+        encoder_hidden_states_t5 = self.text_embedder(
+            encoder_hidden_states_t5.view(-1, encoder_hidden_states_t5.shape[-1])
+        )
+        encoder_hidden_states_t5 = encoder_hidden_states_t5.view(batch_size, sequence_length, -1)
+
+        encoder_hidden_states = torch.cat([encoder_hidden_states, encoder_hidden_states_t5], dim=1)
+        text_embedding_mask = torch.cat([text_embedding_mask, text_embedding_mask_t5], dim=-1)
+        text_embedding_mask = text_embedding_mask.unsqueeze(2).bool()
+
+        encoder_hidden_states = torch.where(text_embedding_mask, encoder_hidden_states, self.text_embedding_padding)
+
+        block_res_samples = ()
+        for layer, block in enumerate(self.blocks):
+
+            hidden_states = block(
+                hidden_states,
+                temb=temb,
+                encoder_hidden_states=encoder_hidden_states,
+                image_rotary_emb=image_rotary_emb,
+            )  # (N, L, D)
+
+            block_res_samples = block_res_samples + (hidden_states, )
+
+            # if layer > self.config.num_layers // 2:
+            #     skip = skips.pop()
+            #     hidden_states = block(
+            #         hidden_states,
+            #         temb=temb,
+            #         encoder_hidden_states=encoder_hidden_states,
+            #         image_rotary_emb=image_rotary_emb,
+            #         skip=skip,
+            #     )  # (N, L, D)
+            # else:
+            #     hidden_states = block(
+            #         hidden_states,
+            #         temb=temb,
+            #         encoder_hidden_states=encoder_hidden_states,
+            #         image_rotary_emb=image_rotary_emb,
+            #     )  # (N, L, D)
+
+            # if layer < (self.config.num_layers // 2 - 1):
+            #     skips.append(hidden_states)
+
+        controlnet_block_res_samples = ()
+        for block_res_sample, controlnet_block in zip(block_res_samples, self.controlnet_blocks):
+            block_res_sample = controlnet_block(block_res_sample)
+            controlnet_block_res_samples = controlnet_block_res_samples + (block_res_sample,)
+
+
+        # 6. scaling
+        controlnet_block_res_samples = [sample * conditioning_scale for sample in controlnet_block_res_samples]
+
+        # if self.config.global_pool_conditions:
+        #     controlnet_block_res_samples = [
+        #         torch.mean(sample, dim=(1, ), keepdim=True) for sample in controlnet_block_res_samples
+        #     ]
+
+
+        if not return_dict:
+            return (controlnet_block_res_samples, )
+        return ControlNetOutput(controlnet_block_samples=controlnet_block_res_samples)
+
+
+class MultiControlNetHunyuanDiT2DModel(ModelMixin):
```

Comment:
can we add a forward method here similarly to what we did in sd3 multi-controlnet?